docs/contributing/design-documents.rst

```diff
@@ -0,0 +1,74 @@
+:title: Design Documents
+:description: Information necessary for a feature design document.
+
+.. _design-documents:
+
+Design Documents
+================
+
+Before submitting a pull request which is expected to significantly change the
+behavior of Deis (such as when contributing a new feature or refactor),
+contributors should first open an issue representing a design document.
+
+Goals
+-----
+
+Design documents were adopted in order to:
+
+* Involve stakeholders as early as possible in a feature's development
+* Ensure code changes accomplish the original motivations and design goals
+* Establish clear acceptance criteria for a feature or change
+* Enforce test-driven design methodology and automated test coverage
+
+Contents
+--------
+
+Design document issues should be named ``Design Doc: <change description>`` and
```

Comment:
I see the general tracking issue being a typically short, high-level description of an issue - such as "we need automated tests for the client". This issue would be assigned a milestone, but wouldn't really be fleshed out in much detail. When someone begins work, they'd open a design document to flesh out exactly how they're going about the implementation of the issue.

I can't really formulate a strong argument as to why it should be a separate issue, other than that it *feels* like it'd get more attention and review. But there are benefits to keeping it in the same issue, such as watchers of that issue being notified when there's an update. I'm open to discussion on this.